homeassistant/components/qnap/config_flow.py

```diff
@@ -0,0 +1,183 @@
+from homeassistant.components.persistent_notification import create as notify_create
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_MONITORED_CONDITIONS,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SSL,
+    CONF_TIMEOUT,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+)
+from homeassistant.core import callback
+from homeassistant.helpers import config_validation as cv
+
+from .const import (
+    _MONITORED_CONDITIONS,
+    CONF_DRIVES,
+    CONF_NICS,
+    CONF_VOLUMES,
+    DEFAULT_PORT,
+    DEFAULT_TIMEOUT,
+)
+from .const import DOMAIN  # pylint:disable=unused-import
+
+NICS_SCHEMA = vol.Schema(
+    {
+        vol.Optional(CONF_NICS): cv.string,
+    }
+)
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+        vol.Optional(CONF_SSL, default=False): cv.boolean,
+        vol.Optional(CONF_VERIFY_SSL, default=True): cv.boolean,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_TIMEOUT, default=DEFAULT_TIMEOUT): cv.positive_int,
+        vol.Optional(CONF_MONITORED_CONDITIONS, default=[]): cv.multi_select(
+            _MONITORED_CONDITIONS
+        ),
+        vol.Optional(CONF_NICS): cv.string,
+        vol.Optional(CONF_DRIVES): cv.string,
+        vol.Optional(CONF_VOLUMES): cv.string,
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class QnapConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Qnap configuration flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    def __init__(self):
+        """Initialize."""
+        self.is_imported = False
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return OptionsFlowHandler(config_entry)
+
+    async def async_step_import(self, import_info):
+        """Set the config entry up from yaml."""
+        self.is_imported = True
+        return await self.async_step_user(import_info)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+        if user_input is not None:
+            host = user_input.get(CONF_HOST)
+            protocol = "https" if user_input.get(CONF_SSL, False) else "http"
+            api = QNAPStats(
+                host=f"{protocol}://{host}",
+                port=user_input.get(CONF_PORT, DEFAULT_PORT),
+                username=user_input.get(CONF_USERNAME),
+                password=user_input.get(CONF_PASSWORD),
+                verify_ssl=user_input.get(CONF_VERIFY_SSL, True),
+                timeout=user_input.get(CONF_TIMEOUT, DEFAULT_TIMEOUT),
+            )
+            try:
+                stats = await self.hass.async_add_executor_job(api.get_system_stats)
+            except Exception:  # noqa: E722 pylint: disable=broad-except
```

Comment:
Couldn't this be less broad ? You are only logging QNAP errors...

Reply:
unfortunately the QNAP API does not handle any errors, so I have to take everything.

Reply:
You should be able to see what errors the API will produce? Is it RequestsError? ValueError?